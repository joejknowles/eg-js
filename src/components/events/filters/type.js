import React from 'react';

const typeOptions = [
  'all',
  'gig',
  'other'
]

export const TypeFilter = (props) => (
  <select onChange={ event => props.onChange(event.target.value) } >
    <option key='all' value=''>Select type</option>
    { typeOptions.map(
      t => <option
        key={ t }
        value={ t }>
        { t }</option>
    ) }
  </select>
);
