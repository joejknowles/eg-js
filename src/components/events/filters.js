import React from 'react';

export const Filters = (props) => (
  <div className='Filters'>
    <input onChange={ event => props.onTypeFilterChange(event.target.value) } ></input>
  </div>
);
