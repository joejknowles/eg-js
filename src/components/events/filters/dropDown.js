import React from 'react';

export const FilterDropDown = (props) => (
  <select onChange={ event => props.onChange(event.target.value) } >
    <option key='all' value=''>Any { props.name }</option>
    { props.options.map(
      t => <option
        key={ t }
        value={ t }>
        { t }</option>
    ) }
  </select>
);
