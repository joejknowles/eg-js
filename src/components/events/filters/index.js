import React from 'react';
import { connect } from 'react-redux';

export const Filters = (props) => (
  <div className='Filters'>
    <select onChange={ event => props.onTypeFilterChange(event.target.value) } >
      <div></div>
    </select>
  </div>
);

const mapStateToDispatch = (dispatch) => ({
  onTypeFilterChange: (filter) => dispatch({ type: 'SET_TYPE_FILTER', filter})
});

export default connect(undefined, mapStateToDispatch)(Filters);
