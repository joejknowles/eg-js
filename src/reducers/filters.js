import { combineReducers } from 'redux';

export const typeReducer = (state = '', action) => {
  if (action.type === 'SET_TYPE_FILTER') {
    return action.filter;
  }
  return state
};

export default combineReducers({ type: typeReducer });
