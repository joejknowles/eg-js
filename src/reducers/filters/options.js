import { combineReducers } from 'redux';

import { locationFromVenue } from '../../selectors/location';

const getUniqueValues = ({
  result, entities: { events }
}, selector) => {
  return result.reduce((accumulator, id) => {
    const value = selector(events[id]);
    if (value && !accumulator.includes(value)) {
      accumulator.push(value);
    }
    return accumulator;
  }, []);
};


const getUniqueTypes = (action) => getUniqueValues(action.response, (event) => event.type);

export const typeOptions = (state = [], action) => {
  if (action.type === 'FETCH_EVENTS_SUCCESS') {
    return getUniqueTypes(action);
  }
  return state;
};

const getUniqueLocations = (action) => (
  getUniqueValues(action.response,
    (event) => locationFromVenue(event.venue || '')
  )
);

export const locationOptions = (state = [], action) => {
  if (action.type === 'FETCH_EVENTS_SUCCESS') {
    const locations = getUniqueLocations(action);
    return locations;
  }
  return state;
};

export default combineReducers({ typeOptions, locationOptions });
