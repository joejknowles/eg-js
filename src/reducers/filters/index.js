import { combineReducers } from 'redux';

import options from './options'

export const typeReducer = (state = '', action) => {
  if (action.type === 'SET_TYPE_FILTER') {
    return action.filter;
  }
  return state
};

export const location = (state = '', action) => {
  if (action.type === 'SET_LOCATION_FILTER') {
    return action.filter;
  }
  return state
};

export const titleSearch = (state = '', action) => {
  if (action.type === 'SET_TITLE_SEARCH') {
    return action.term;
  }
  return state
};

export default combineReducers({ type: typeReducer, location, titleSearch, options });
