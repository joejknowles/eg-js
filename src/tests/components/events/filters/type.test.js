import { TypeFilter } from '../../../../components/events/filters/type';

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';

const shallowRenderer = ReactTestUtils.createRenderer();

it('renders type filter as before', () => {
  const component = shallowRenderer.render(<TypeFilter />);
  expect(component).toMatchSnapshot();
});
