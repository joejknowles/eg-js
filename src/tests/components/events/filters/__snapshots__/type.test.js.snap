exports[`test renders type filter as before 1`] = `
<select
  onChange={[Function]}>
  <option
    value="">
    Select type
  </option>
  <option
    value="gig">
    gig
  </option>
  <option
    value="other">
    other
  </option>
</select>
`;
